src/main/scala/intellij/haskell/module/HaskellModuleBuilder.scala

```diff
@@ -221,9 +223,12 @@ object HaskellModuleBuilder {
   private def downloadHaskellPackageSources(project: Project, stackPath: String, haskellPackages: Seq[HaskellPackageInfo], progressIndicator: ProgressIndicator) = {
     val step = 0.8 / haskellPackages.size
     var progressFraction = InitialProgressStep
-    haskellPackages.flatMap { packageInfo =>
-      val fullName = packageInfo.name + "-" + packageInfo.version
-      val stdErr = CommandLine.runProgram(Some(project), project.getBasePath + File.separator + LibName, stackPath, Seq("unpack", fullName), 10000, captureOutputToLog = true, logErrorAsInfo = true).map(_.getStderr)
+    haskellPackages.filter(packageInfo => {
```

Comment:
What do you mean by `finding modules is not going work anymore because module names are not unique anymore.`? can you give me more details? 😄 
>I think your last two PR's are intermingled.
Also packages which are nog used anymore have to be deleted otherwise finding modules is not going work anymore because module names are not unique anymore.